### ISSUE TITLE
Cannot detect react component if JSX is returned as variable

### ISSUE THREAD

**Issue by danielcooke1996** [yannickcr/eslint-plugin-react#2379]
The following component will not trigger any eslint errors with this plugin installed:

```javascript
const Foo = ({ bar }) => {
  const baz = <div>Hello World</div>;

  return baz;
};
```

**Expected Result**
`react/prop-types` error

**Actual Result**
No prop-type error

**Version**
eslint: 6.1.0
eslint-plugin-react: 7.14.3
react: 16.8.6

**Comment by ljharb** [comment]
Indeed, there’s no way to reliably statically detect an SFC that doesn’t return jsx. Returning the fragment is a good workaround.
